tt_metal/include/tt_metal/command_queue.hpp

```diff
@@ -0,0 +1,78 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "tt_metal/tt_stl/span.hpp"
+#include "types.hpp"
+
+//==================================================
+//                COMMAND QUEUE OPERATIONS
+//==================================================
+
+namespace tt::tt_metal {
+namespace v1 {
+
+/**
+ * @brief Reads a buffer from the device.
+ *
+ * @param cq The command queue used to dispatch the command.
+ * @param buffer The device buffer to read from.
+ * @param dst Destination memory to copy from device to host.
+ * @param blocking Indicates whether the operation is blocking.
+ */
+void EnqueueReadBuffer(CommandQueueHandle cq, Buffer buffer, stl::Span<std::byte> dst, bool blocking);
```

Comment:
Along the lines of `ReadFromBuffer`, should this potentially return a `std::future<std::size_t>` whose result is the number of bytes read?